staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -618,16 +619,16 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 		if err != nil {
 			return err
 		}
-		patchBytes, patchedObject, err := patcher.Patch(info.Object, modified, info.Source, info.Namespace, info.Name, o.ErrOut)
+		patchedObject, err := patcher.Patch(info.Object, modified, info.Source, info.Namespace, info.Name, o.ErrOut)
```

Comment:
What is really the goal of having patched object here? Just so that we can see if it's empty? (and to do `info.Refresh(patchedObject, true)`). I'm curious if we can remove that. It sounds to me like the `Patch` operation should be deciding if the operation was "unchanged", rather than trying to guess later.

Reply:
First of all, while looking at the code after ^ suggestion. I found a major bug. This PR checks equality **after** `info.refresh` which should cause every patching "unchanged". I don't know how tests did not catch it. cc: @alexzielenski

Reply:
Primary reason is `diff` command uses it as [merged object](https://github.com/kubernetes/kubernetes/blob/47a1d1dfe329301353030c1d2c1a5b5d3f2d2a55/staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go#L399.). If we remove `patchObjects`, diff command will need a lot of changes.

In addition to that, `info.Refresh`, which is also used for SSA, does not just copy objects. It has [some custom logic](https://github.com/kubernetes/kubernetes/blob/a5aa858d44651ba48bdce634a39b55be91216614/staging/src/k8s.io/cli-runtime/pkg/resource/visitor.go#L118)(currently this logic looks standard in various cases, patch, refresh object, etc.)

I think, returning `patchObject` by "Patch" operation sounds reasonable. It returns final patched object and lets callers do whatever they intend to do. Because for example, `diff` command does not care about object has changed or not.

Reply:
I haven't dug too deep into questioning do we need to return `patchedObject`, but it seems reasonable and now with the `patchBytes` removed to return a boolean informing if the patch modified the object or not. It will allow us to drop several conditions throughout the code base.

Reply:
@apelisse @soltysh I also removed `patchedObject`. We are also refreshing `info.Object` in `Patch` function.